pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -26,6 +22,38 @@
 from ..pyOpt_utils import ICOL, IDATA, INFINITY, IROW, extractRows, mapToCSC, scaleRows
 
 
+# Compiled module
+def _import_snopt(absolute: bool = False, path_prepend: Optional[str] = None):
+    """Helper to load snopt bindings."""
+    if absolute:
+        if path_prepend:
+            sys.path.insert(0, path_prepend)
+        import snopt  # isort: skip
+
+        if path_prepend:
+            sys.path.remove(path_prepend)
+    else:
+        from . import snopt  # isort: skip
+    return snopt
+
+
+_SNOPT_PATH_PREPEND = os.environ.get("PYOPTSPARSE_SNOPT_PATH_PREPEND", None)
+if _SNOPT_PATH_PREPEND is not None:
+    try:
+        snopt = _import_snopt(absolute=True, path_prepend=_SNOPT_PATH_PREPEND)
+    except ImportError:
```

Comment:
I think it could be helpful to add here a warning message to make the user aware that `snopt` could not be imported from `PYTHONPATH`  - most likely `PYOPTSPARSE_SNOPT_PATH_PREPEND` is incorrectly set.